Frontend/src/components/calendar/CalendarModal.jsx

```diff
@@ -0,0 +1,138 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { UserContext } from '../../UserContext.jsx';
+import { useNavigate } from 'react-router-dom';
+import '../mentor/mentor-profile/MentorProfileModal.css'
+import config from '../../../config.js';
+
+function CalendarModal ({toggleModal, onMeetingScheduled}) {
+    const {user} = useContext(UserContext);
+    const [menteesOrMentors, setMenteesOrMentors] = useState([]);
+    const [selectedUser, setSelectedUser] = useState('');
+    const [scheduledDate, setScheduledDate] = useState('');
+    const [scheduledTime, setScheduledTime] = useState('');
+    const [topic, setTopic] = useState('');
+    const [errorMessage, setErrorMessage] = useState('');
+    const [userData, setUserData] = useState('');
+    const navigate = useNavigate();
+
+    // Fetch user data (mentor or mentee)
+    useEffect(() => {
+        const fetchUserData = async () => {
+            try {
+                const url = user.userRole === 'mentor'
+                    ? `${config.apiBaseUrl}/mentors/${user.id}`
+                    : `${config.apiBaseUrl}/mentees/${user.id}`;
+
+                const response = await fetch(url);
+                if (!response.ok) {
+                    throw new Error('Failed to fetch user data');
+                }
+                const data = await response.json();
+                setUserData(user.userRole === 'mentor' ? data.mentor : data.mentee);
```

Comment:
Looks like checking if a user is a mentor or mentee is very common. Would make sense to encapsulate that logic into a helper function

Reply:
Resolved